spec/models/publish_services_event_metadata_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe PublishServicesMetadataEvent, type: :model do
+  publish_date = Time.now
+  event_id = 0
+  component = Component.create(name: 'lala', component_type: 'MSA')
+  let(:event) do
+    service_metadata = ServicesMetadata.to_json(event_id, [component], publish_date)
+    PublishServicesMetadataEvent.create(
+      event_id: event_id, cert_config: service_metadata
+    )
+  end
+
+  context '#create' do
+    it 'creates an event which contains hard-coded data' do
+      expect(event.data).to include(
+        'event_id',
+        'cert_config'
+      )
+      expect(event.data['cert_config']).to include('publish_date')
+    end
+    
+    it 'can attach config metadata' do
+      event.upload
+      expect(event.document.attached?).to be true
+      expect(ActiveStorage::Blob.service.exist?(event.document.key)).to be true
+    end
+
+    it 'generates a link to uploaded config metadata' do
```

Comment:
This seems interesting - can you give a demo of this?

Reply:
yes